File: backend/app/schemas/notifications.py
"""Read-only contracts for private in-app notification records and preferences."""

from datetime import datetime

from pydantic import BaseModel

from app.models.entities import NotificationLevel


class NotificationResponse(BaseModel):
    id: str
    case_id: str | None
    category: str
    level: NotificationLevel
    title: str
    body: str | None
    read_at: datetime | None
    created_at: datetime


class NotificationPreferenceResponse(BaseModel):
    id: str
    category: str
    in_app_enabled: bool
    created_at: datetime
    updated_at: datetime


class NotificationPreferenceUpdateRequest(BaseModel):
    in_app_enabled: bool

File: frontend/client/src/api/notifications.ts
import { apiClient } from "./client";

export type InAppNotificationRecord = { id: string; case_id: string | null; category: string; level: string; title: string; body: string | null; read_at: string | null; created_at: string };
export type NotificationPreferenceRecord = { id: string; category: string; in_app_enabled: boolean; created_at: string; updated_at: string };

export async function getNotifications(caseId?: string): Promise<InAppNotificationRecord[]> { return (await apiClient.get("/notifications", { params: caseId ? { case_id: caseId } : undefined })).data; }
export async function getNotificationPreferences(): Promise<NotificationPreferenceRecord[]> { return (await apiClient.get("/notifications/preferences")).data; }
